web/src/reactions.js

```diff
@@ -87,7 +93,9 @@ function update_ui_and_send_reaction_ajax(message_id, reaction_info) {
     const args = {
         url: "/json/messages/" + message_id + "/reactions",
         data: reaction_info,
-        success() {},
+        success() {
+            view.loading_reactions_ids[local_id] = false;
+        },
         error(xhr) {
             const response = channel.xhr_error_message("Error sending reaction", xhr);
```

Comment:
Good catch, thanks!

Reply:
It looks like PR #4291 has been merged and PR #4295 has been closed, so we don't need them anymore. Additionally, if there are fewer race condition errors, I think we could change the `bluesleep.warn`  to the `bluesleep.error` now and remove the comment entirely.

Reply:
Yeah, those PRs are ancient!  Thanks for doing the cleanup here.

Though that makes me think -- since we have a local echo of the reactji, is there anything we need to do to prevent wonkiness on the _client_ side here, when we call `add_reaction` twice?  Or is the second one a no-op?

Reply:
The modifications made will also ensure that the local echo remains unchanged, thus ensuring that everything on the client side operates correctly. This matter was previously discussed [here].(https://chat.zulip.org/#narrow/stream/2-general/topic/Emoji.20reactions.20parallel.20request.20race/near/1517766)